hazelcast/src/test/java/com/hazelcast/test/starter/AbstractStarterObjectConstructor.java

```diff
@@ -38,10 +39,9 @@ public Object createNew(Object arg) {
         try {
             return createNew0(arg);
         } catch (Exception e) {
-            throw ExceptionUtil.rethrow(e);
+            throw rethrow(e);
```

Comment:
Should this rethrow a `GuardianException`?

Reply:
Probably we should check here for the class loader of the exception class, if it's the class loader hosting the test cases it should be `GuardianException`, otherwise we should transfer the exception using `com.hazelcast.test.starter.Utils#transferThrowable`.

Reply:
`throw new GuardianException(transferThrowable(e));` seems to be the easiest way to combine this, since a pure `Throwable` cannot be rethrown here.

Reply:
Another option is to replace `transferThrowable` with something like this https://gist.github.com/taburet/9577c46265529f55c134b49b2379e249